modal/_function_utils.py

```diff
@@ -86,8 +86,17 @@ class FunctionInfo:
     def __init__(self, f, serialized=False, name_override: Optional[str] = None, cls: Optional[Type] = None):
         self.raw_f = f
         self.cls = cls
-        # TODO(erikbern): if f.__qualname__ != f.__name__,  we should infer the class name instead
-        self.function_name = name_override if name_override is not None else f.__qualname__
+
+        if name_override is not None:
+            self.function_name = name_override
+        elif f.__qualname__ != f.__name__:
+            # Class function.
+            if len(f.__qualname__.split(".")) > 2:
```

Comment:
maybe we should fallback to `__qualname__` here if len > 2? I don't think anyone was legitimately using that though